### ISSUE TITLE
pa11y-ci uses exit code 0 when a URL "failed to run"

### ISSUE THREAD

**Issue by jpmckinney** [pa11y/pa11y-ci#228]
I would expect it to use a non-zero exit code.

If there is a good reason for this, could an option be added to change this behavior?

**Comment by bonkrat** [comment]
I agree with the above in that it might make sense to use a non-zero exit code when the tool fails to run - this would have made it more clear that a CI pipeline using `pa11y-ci` was not working as intended.

For anyone that stumbles here trying to resolve the issue... I needed to [pass `--no-sandbox` to my configuration](https://github.com/pa11y/pa11y-ci?tab=readme-ov-file#docker) as I am `pa11y-ci` in a Docker container.

**Comment by jpmckinney** [comment]
This was back on pa11y 6.2.3 and pa11y-ci 3.1.0.

Once I got things working, my command was:

```bash
npx pa11y-ci -c pa11y.default.json -s http://127.0.0.1:8000/sitemap.xml
```

where the config was:

```json
{
  "defaults": {
    "chromeLaunchConfig": {
      "args": [
        "--no-sandbox"
      ]
    },
    "ignore": [
      "WCAG2AA.Principle3.Guideline3_2.3_2_2.H32.2"
    ]
  }
}
```

I don't remember what caused the "failed to run" error. Maybe pa11y handles some technical errors better now. I have no commits before getting it working, so I can't discover what caused the error.

Using those old versions, pa11y-ci does return error code 2 if, e.g.:

- one URL passes
- another URL is unresolvable (e.g. `Error: net::ERR_NAME_NOT_RESOLVED at https://false`)

And it does error in CI if `--no-sandbox` is missing.

I don't know all the cases in which error() is called and prints the "failed to run", but I assume there was at least one such case and it somehow didn't lead to error code 2.

But, maybe it's all fixed.

**Comment by aarongoldenthal** [comment]
Thanks for the additional information @jpmckinney.

> And it does error in CI if `--no-sandbox` is missing.

Just to be clear, if "in CI" is in GitHub Actions and you're using `--no-sandbox` (not within a container) I'd highly recommend reading [this description of the problem and various fixes](https://chromium.googlesource.com/chromium/src/+/main/docs/security/apparmor-userns-restrictions.md). While `--no-sandbox` will resolve the issue in this case, it is the least secure of the various options, so you might want to consider others (for example, the `pa11y`/`pa11y-ci` tests use [option 1](https://github.com/pa11y/pa11y-ci/blob/f4d28387b2b7a866454a4a41a7ba2e98f9a46201/.github/workflows/tests.yml#L35-L39).

That probably needs a more prominent example in the `pa11y-ci` docs (although in CI, testing against known sites, the risk is low).
